flink-table/flink-table-api-java/src/main/java/org/apache/flink/table/api/internal/TableEnvironmentImpl.java

```diff
@@ -1108,11 +1109,20 @@ private TableResult executeOperation(Operation operation) {
             return buildShowResult("table name", listTables());
         } else if (operation instanceof ShowFunctionsOperation) {
             ShowFunctionsOperation showFunctionsOperation = (ShowFunctionsOperation) operation;
-            return buildShowResult(
-                    "function name",
-                    showFunctionsOperation.requireUser()
-                            ? listUserDefinedFunctions()
-                            : listFunctions());
+            String[] functionNames = null;
+            ShowFunctionsOperation.FunctionScope functionScope =
+                    showFunctionsOperation.getFunctionScope();
+            switch (functionScope) {
+                case USER:
+                    functionNames = listUserDefinedFunctions();
+                    break;
+                case ALL:
+                    functionNames = listFunctions();
+                    break;
+                default:
+                    throw new UnsupportedFunctionScopeException(functionScope.toString());
```

Comment:
We don't need to introduce a new Exception class. Can simply use `new UnsupportedOperationException(String.format("SHOW FUNCTIONS with %s scope is not supported.", scope))` here .

Reply:
Yes